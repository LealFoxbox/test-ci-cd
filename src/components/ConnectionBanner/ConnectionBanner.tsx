import React from 'react';
import { Text, useTheme } from 'react-native-paper';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

import { styled, withTheme } from 'src/paperTheme';

const Container = withTheme(
  styled.View`
    background-color: ${({ theme }) => theme.colors.accent};
    flex-direction: row;
    align-items: center;
    justify-content: center;
  `,
);

const ColoredText = withTheme(
  styled(Text)`
    color: ${({ theme }) => theme.colors.background};
    padding: 10px;
    text-align: center;
  `,
);

interface ConnectionBannerProps {
  connected: boolean;
}

const ConnectionBanner: React.FC<ConnectionBannerProps> = ({ connected }) => {
  const theme = useTheme();

  if (connected) {
    return null;
  }

  return (
    <Container>
      <MaterialIcons name="signal-wifi-off" color={theme.colors.background} size={26} />
      <ColoredText>No internet connection</ColoredText>
    </Container>
  );
};

export default ConnectionBanner;
