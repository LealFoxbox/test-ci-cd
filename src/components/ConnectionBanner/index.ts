export * from './ConnectionBanner';
export { default } from './ConnectionBanner';
