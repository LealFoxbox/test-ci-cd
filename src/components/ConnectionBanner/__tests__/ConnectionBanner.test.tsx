import React from 'react';
import { render } from '@testing-library/react-native';

import ConnectionBanner from '../ConnectionBanner';

describe('ConnectionBanner', () => {
  it('should render', () => {
    const result = render(<ConnectionBanner connected={false} />);

    expect(result).toMatchSnapshot();
  });
});
