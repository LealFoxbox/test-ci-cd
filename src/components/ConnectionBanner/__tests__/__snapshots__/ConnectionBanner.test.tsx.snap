// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionBanner should render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#03dac4",
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
  theme={
    Object {
      "animation": Object {
        "scale": 1,
      },
      "colors": Object {
        "accent": "#03dac4",
        "backdrop": "rgba(0, 0, 0, 0.5)",
        "background": "#f6f6f6",
        "disabled": "rgba(0, 0, 0, 0.26)",
        "error": "#B00020",
        "notification": "#f50057",
        "onBackground": "#000000",
        "onSurface": "#000000",
        "placeholder": "rgba(0, 0, 0, 0.54)",
        "primary": "#6200ee",
        "surface": "#ffffff",
        "text": "#000000",
      },
      "dark": false,
      "fonts": Object {
        "light": Object {
          "fontFamily": "System",
          "fontWeight": "300",
        },
        "medium": Object {
          "fontFamily": "System",
          "fontWeight": "500",
        },
        "regular": Object {
          "fontFamily": "System",
          "fontWeight": "400",
        },
        "thin": Object {
          "fontFamily": "System",
          "fontWeight": "100",
        },
      },
      "roundness": 4,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#f6f6f6",
          "fontSize": 26,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î‡š
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "#f6f6f6",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "textAlign": "center",
          },
        ],
      ]
    }
  >
    No internet connection
  </Text>
</View>
`;
